src/hooks/useSelectChain.ts

```diff
@@ -16,12 +17,17 @@ export default function useSelectChain() {
     async (targetChain: SupportedChainId) => {
       if (!connector) return
 
-      const connectionType = getConnection(connector).type
+      const connection = getConnection(connector)
+      const connectionType = connection.type
 
       try {
         dispatch(updateConnectionError({ connectionType, error: undefined }))
         await switchChain(connector, targetChain)
       } catch (error) {
+        if (didUserReject(connection, error)) {
```

Comment:
If the user did not reject, but the chain change failed, shouldn't we still show the error state?

Reply:
A toast should pop up which is what happens here